core/primitives/src/views.rs

```diff
@@ -211,7 +211,8 @@ pub struct StateItem {
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct ViewStateResult {
     pub values: Vec<StateItem>,
-    pub proof: TrieProofPath,
+    // signals whether the value was found in the trie, and the proof itself to verify the claim
+    pub proof: Option<(bool, TrieProofPath)>,
```

Comment:
I haven't followed all the discussion on this PR in detail.  Are we sure we are happy with using a `bool` here and below to signify membership or should we consider introducing an enum for it?

Reply:
I've added `ProofState`